ccl/streamingccl/streamingest: TestTenantStreamingMultipleNodes failed

ccl/streamingccl/streamingest.TestTenantStreamingMultipleNodes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10084283?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10084283?buildTab=artifacts#/) on master @ [a65bca232dfde1862e5aff343e7b1f3a379d6cd6](https://github.com/cockroachdb/cockroach/commits/a65bca232dfde1862e5aff343e7b1f3a379d6cd6):


```
=== RUN   TestTenantStreamingMultipleNodes
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingMultipleNodes3104947643
    test_log_scope.go:79: use -show-logs to present logs inline
    testutils.go:342: condition failed to evaluate within 45s: stream ingestion has not recorded any progress yet, waiting to advance pos 1684133150.612214495,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantStreamingMultipleNodes3104947643
--- FAIL: TestTenantStreamingMultipleNodes (365.20s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #101281 ccl/streamingccl/streamingest: TestTenantStreamingMultipleNodes failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1.0]
- #99900 ccl/streamingccl/streamingest: TestTenantStreamingMultipleNodes failed [C-test-failure O-robot T-disaster-recovery branch-release-23.1]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingMultipleNodes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27914

--- a/pkg/ccl/streamingccl/replicationtestutils/BUILD.bazel
+++ b/pkg/ccl/streamingccl/replicationtestutils/BUILD.bazel
@@ -37,6 +37,7 @@ go_library(
         "//pkg/testutils",
         "//pkg/testutils/jobutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/contextutil",

--- a/pkg/ccl/streamingccl/replicationtestutils/testutils.go
+++ b/pkg/ccl/streamingccl/replicationtestutils/testutils.go
@@ -33,6 +33,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/jobutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
@@ -339,7 +340,11 @@ func (c *TenantStreamingClusters) WaitUntilStartTimeReached(ingestionJobID jobsp
 }
 
 func WaitUntilStartTimeReached(t *testing.T, db *sqlutils.SQLRunner, ingestionJobID jobspb.JobID) {
-	testutils.SucceedsSoon(t, func() error {
+	timeout := 45 * time.Second
+	if skip.NightlyStress() {
+		timeout *= 3
+	}
+	testutils.SucceedsWithin(t, func() error {
 		payload := jobutils.GetJobPayload(t, db, ingestionJobID)
 		details, ok := payload.Details.(*jobspb.Payload_StreamIngestion)
 		if !ok {
@@ -364,7 +369,7 @@ func WaitUntilStartTimeReached(t *testing.T, db *sqlutils.SQLRunner, ingestionJo
 				highwater.String(), startTime.String())
 		}
 		return nil
-	})
+	}, timeout)
 }
 
 func CreateScatteredTable(t *testing.T, c *TenantStreamingClusters, numNodes int) {

